numba/npyufunc/_internal.c

```diff
@@ -104,6 +105,557 @@ PyTypeObject PyUFuncCleaner_Type = {
     0,                                          /* tp_version_tag */
 };
 
+/* ______________________________________________________________________
+ * DUFunc: A call-time (hence dynamic) specializable ufunc.
+ */
+
+typedef struct {
+    PyObject_HEAD
+    PyObject      * dispatcher;
```

Comment:
Ah! If it's only used in the derived class, then, it should probably be a member of the derived class. No need to maintain it at the C level.